src/wincameranodelet.cpp

```diff
@@ -0,0 +1,121 @@
+#include "wincapture.h"
+#include "winrospublisher.h"
+#include <ros/ros.h>
+#include <string>
+#include <nodelet/nodelet.h>
+
+namespace ros_win_camera
+{
+    const int32_t PUBLISHER_BUFFER_SIZE = 4;
+
+    class WinCameraNodelet : public nodelet::Nodelet
+    {
+
+    public:
+        WinCameraNodelet()
+        {
+        }
+        ~WinCameraNodelet()
+        {
+            waitForFinish.lock();
+        }
+
+    private:
+        virtual void onInit()//int main(int argc, char** argv)
+        {
+            //ros::init(argc, argv, "win_camera");
+            //ros::NodeHandle private_node("~");
+            auto privateNode = getPrivateNodeHandle();
+            std::string videoSourcePath = "";
+            bool isDevice = true;
+            float frameRate = 30;
+            winrt::com_ptr< ros_win_camera::WindowsMFCapture> camera, camera1;
+            std::string frame_id("camera");
+            privateNode.getParam("frame_rate", frameRate);
+
+            winrt::slim_mutex waitForFinish;

```

Comment:
nit: can we use standard library thread primitives here? https://github.com/microsoft/xlang/issues/476